vivarium/framework/population.py

```diff
@@ -178,21 +190,60 @@ def get_view(self, columns, query=None):
         """
         return PopulationView(self, columns, query)
 
-    def get_simulant_creator(self):
+    def register_simulant_initializer(self, initializer: Callable,
+                                      creates_columns: Sequence[str]=(), requires_columns: Sequence[str]=()):
+        self._population_initializers.append((initializer, creates_columns, requires_columns))
```

Comment:
should this `raise` if `_initializers_ordered`?

Reply:
There's an implicit contract that there won't be `_initializers_ordered` when this method gets called.  This method is only accessible from the `Builder` object that gets passed into each component's setup method.  `_initializers_ordered` only gets populated the first time we create simulants, an event guaranteed by the framework to occur after the setup phase.